src/cli/cli_coap.cpp

```diff
@@ -231,6 +355,15 @@ otError Coap::ProcessRequest(int argc, char *argv[])
     messageInfo.mPeerAddr = coapDestinationIp;
     messageInfo.mPeerPort = OT_DEFAULT_COAP_PORT;
 
+    if (coapObserve)
+    {
+        // Make a note of the message details for later so we can cancel it later.
+        memcpy(&mRequestAddr, &coapDestinationIp, sizeof(mRequestAddr));
+        mRequestTokenLength = otCoapMessageGetTokenLength(message);
+        memcpy(mRequestToken, otCoapMessageGetToken(message), mRequestTokenLength);
+        strlcpy(mRequestUri, coapUri, kMaxUriLength);
```

Comment:
Prefer using `strncpy` - we are trying to move away from `strlpy` (see https://github.com/openthread/openthread/commit/708c250fe5ef7057e6395d9759fb341d150e89e8).